libs/shared/form-fields/src/lib/PhoneInputController/PhoneInputController.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useState, forwardRef } from 'react'
+import {
+  InputBackgroundColor,
+  Option,
+  PhoneInput,
+} from '@island.is/island-ui/core'
+import {
+  Controller,
+  Control,
+  ValidationRules,
+  useFormContext,
+} from 'react-hook-form'
+import NumberFormat from 'react-number-format'
+import { TestSupport } from '@island.is/island-ui/utils'
+import { ValueType } from 'react-select'
+import { countryCodes as countryCodeList } from './countryCodes'
+
+interface Props {
+  autoFocus?: boolean
+  defaultValue?: string
+  disabled?: boolean
+  control?: Control
+  rules?: ValidationRules
+  error?: string
+  id: string
+  label?: string
+  name?: string
+  onChange?: (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => void
+  placeholder?: string
+  backgroundColor?: InputBackgroundColor
+  currency?: boolean
+  required?: boolean
+  readOnly?: boolean
+  rightAlign?: boolean
+  loading?: boolean
+  size?: 'xs' | 'sm' | 'md'
+  autoComplete?: 'off' | 'on'
+  allowedCountryCodes?: string[]
+}
+
+interface ChildParams {
+  value?: string
+  onBlur: () => void
+  onChange: (...event: any[]) => void
+  name: string
+}
+
+const DEFAULT_COUNTRY_CODE = '+354'
+
+const getCountryCodes = (allowedCountryCodes?: string[]) => {
+  return countryCodeList
+    .filter((x) =>
+      allowedCountryCodes ? allowedCountryCodes.includes(x.code) : true,
+    )
+    .map((x) => ({
+      label: `${x.name} ${x.dial_code}`,
+      value: x.dial_code,
+      description: x.flag,
+    }))
+}
+
+/**
+ * Gets default value for the controller.
+ * If the incoming value is empty or starts with "+",
+ * then we don't have to prefix it with the country code.
+ */
+const getDefaultValue = (
+  defaultValue?: string,
+  defaultCountryCode?: string,
+) => {
+  return defaultValue === '' || defaultValue?.startsWith('+')
+    ? defaultValue
+    : `${defaultCountryCode ?? ''}${defaultValue}`
+}
+
+/**
+ * Gets default country code.
+ * This function tries to extract a country calling code from a phone number,
+ * and it assumes that the phone number itself is always 7 characters
+ * Might need to rethink this logic in the future when more country calling codes are allowed.
+ *
+ * Example outputs:
+ * getDefaultCountryCode("+3545812345") // +354
+ * getDefaultCountryCode("+455812345") // +45
```

Comment:
If I'm not mistaken a phone number from DK (+45) is 8 digits so this would probably fail. same with many other country codes

Reply:
Yeah this will only work for icelandic phone numbers as it is

The problem i'm facing is how to correctly parse the country code without knowing how long the phone number is.
e.g. +45689912346 could be parsed as country code +45 or +456

I guess the best way would just be to hardcode this information in the data

Reply:
Should loop through countryCodes to find a matching country code. The phone number system is designed so there won't be conflicts like +45 and +456.